### PR TITLE
Disable DTD parsing in all external source plugins

### PR DESCRIPTION
**Pull request by tdonohue**
## References
Fixes a number of _potential_ XXE vulnerabilities (reported by GitHub's code scanning) in our codebase with regards to importing metadata from external sources (PubMed, Web of Science, etc).  

Keep in mind, the risk of these vulnerabilities being exploited is extremely low since these external sources are all trusted APIs.  That said, we obviously should patch our codebase to avoid a scenario where, if the trusted API is exploited, DSpace could also be exploited.

## Description
Per recommendations of OWASP, patched our XML parsing to disable DTD (except in one scenario, PubMed, as noted below).

The patched code was borrowed directly from the OWASP cheat sheet: https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html

In all scenarios, DTDs are disabled **except** for PubMed (`PubMedImportMetadataSourceServiceImpl`).  When DTDs were fully disabled, errors occurred with the PubMed import process.  So, I went with the second best approach from OWASP which was to disallow external entities & entity expansion.  (Somewhat oddly PubMed Europe works fine with DTDs disabled, this only occurred for PubMed)

## Instructions for Reviewers
* Review code changes
* Attempt a new Import from these external sources (these were all patched).  These imports can all be done from MyDSpace:
    * Import -> Person -> ORCID
    * Import -> Item/Publication -> PubMed
    * Import -> Item/Publication -> CiNii  (requires free api key) 
    * Import -> Item/Publication -> European Patent Office (EPO) (requires free api key) 
    * Import -> Item/Publication -> PubMed Europe
    * Import -> Item/Publication -> Scopus (requires free api key) 
    * Import -> Item/Publication -> Web Of Science (*) (UNFORTUNATELY, I cannot test this one. Requires a **paid** api key.)
* NOTE: You do **not** need to complete a submission for each of these scenarios. You just need to verify that the search returns results from the external service & you can click on a result to *start a submission*.  That will verify that XML parsing of the API is unaffected by the patching.
* For those sources that require an API key, see our [documentation](https://wiki.lyrasis.org/pages/viewpage.action?pageId=104566672#ImportingItemsviabasicbibliographicformats(Endnote,BibTex,RIS,CSV,etc)andonlineservices(arXiv,PubMed,CrossRef,CiNii,etc)-SupportedExternalSources)

I have successfully tested all of the modified sources **except** for Web of Science.  I do not have access to a paid API key, and they have been unwilling to provide me with a free one.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/external/provider/orcid/xml/Converter.java [hunk 1]
```diff
@@ -28,11 +31,16 @@
 
     protected Object unmarshall(InputStream input, Class<?> type) throws SAXException, URISyntaxException {
         try {
+            XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+            // disallow DTD parsing to ensure no XXE attacks can occur
+            xmlInputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+            XMLStreamReader xmlStreamReader = xmlInputFactory.createXMLStreamReader(input);
```
[the remark this answers is not in the document]
Comment (tdonohue) [reply dspace-api/src/main/java/org/dspace/external/provider/orcid/xml/Converter.java:37]: The github code scanner has reported an obvious false positive.  The line just above this one is already disabling DTD parsing via the OWASP recommended approach: https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html#xmlinputfactory-a-stax-parser

**Comment by mwoodiupui** [comment]
Agreeing in principle:  "trusted API" can also be pronounced "attractive intermediate target."  "Trusted" is not enough; we should always ask, "trusted for what purpose(s)?"

**Comment by alanorth** [comment]
@tdonohue I tested importing an item from PubMed using ID ` 29700092`. The metadata is retrieved and I was able to start a submission successfully. Looks good to me.

**Comment by tdonohue** [comment]
@benbosman or @artlowel :  Is there someone at Atmire who could give this a quick test (specifically/especially the "Web of Science" plugin after these changes)?  This has been stalled for weeks and I'd like to ensure we get these fixes into 7.6.

**Comment by tdonohue** [comment]
To help testers, I've rebased this on latest `main` (as of today).

**Review by atarix83 — APPROVED** [review]
@tdonohue 

I've tested web of science and it works properly

**Pull request merged**